packages/firebase_storage/firebase_storage/test/firebase_storage_test.dart

```diff
@@ -132,6 +132,31 @@ void main() {
         verify(kMockStoragePlatform.ref(testPath));
       });
 
+      test('verify delegate method is called for http urls with spaces', () {
+        const String customBucket = 'test.appspot.com';
+        const String testPath = 'file with  spaces   .gif';
+        const String url =
+            'https://firebasestorage.googleapis.com/v0/b/$customBucket/o/$testPath?alt=media';
+
+        final ref = storage.refFromURL(url);
+
+        expect(ref, isA<Reference>());
```

Comment:
Shouldn't this also make sure `ref.name` and `ref.fullPath` are actually parsed correctly into the ref?

Reply:
The calls to `name`/`fullPath` etc are from the delegate which is mocked, so the testing for that should be happening in the platform interface code (will check). This test confirms the platform interface was called with a `ref()` of the testPath, which it does so it's safe to assume the parsing is correct.

Reply:
added some tests for the full interface to delegate checks, along with testing the URL utils directly.